homeassistant/components/purpleair/sensor.py

```diff
@@ -0,0 +1,221 @@
+        value_fn=lambda sensor: sensor.pm5_0_um_count,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="pressure",
+        name="Pressure",
+        device_class=SensorDeviceClass.PRESSURE,
+        native_unit_of_measurement=PRESSURE_MBAR,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pressure,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="rssi",
+        name="RSSI",
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        native_unit_of_measurement=SIGNAL_STRENGTH_DECIBELS_MILLIWATT,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.pressure,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="temperature",
+        name="Temperature",
+        device_class=SensorDeviceClass.TEMPERATURE,
+        native_unit_of_measurement=UnitOfTemperature.FAHRENHEIT,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.temperature,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="uptime",
+        name="Uptime",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        icon="mdi:timer",
+        native_unit_of_measurement=TIME_MINUTES,
+        state_class=SensorStateClass.TOTAL_INCREASING,
+        value_fn=lambda sensor: sensor.uptime,
+    ),
+    PurpleAirSensorEntityDescription(
+        key="voc",
+        name="VOC",
+        device_class=SensorDeviceClass.VOLATILE_ORGANIC_COMPOUNDS,
+        native_unit_of_measurement=CONCENTRATION_IAQ,
+        state_class=SensorStateClass.MEASUREMENT,
+        value_fn=lambda sensor: sensor.voc,
+    ),
+]
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up PurpleAir sensors based on a config entry."""
+    coordinator: PurpleAirDataUpdateCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(
+        PurpleAirSensorEntity(coordinator, entry, sensor_index, description)
+        for sensor_index in entry.options[CONF_SENSOR_INDICES]
+        for description in SENSOR_DESCRIPTIONS
+    )
+
+
+class PurpleAirSensorEntity(PurpleAirEntity, SensorEntity):
+    """Define a representation of a PurpleAir sensor."""
+
+    entity_description: PurpleAirSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: PurpleAirDataUpdateCoordinator,
+        entry: ConfigEntry,
+        sensor_index: int,
+        description: PurpleAirSensorEntityDescription,
+    ) -> None:
+        """Initialize."""
+        super().__init__(coordinator, entry, sensor_index)
+
+        self._attr_unique_id = f"{self._sensor_index}-{description.key}"
+        self.entity_description = description
+
+    @property
+    def native_value(self) -> StateType:
```

Comment:
Does this really return everything in `StateType`?

Reply:
Roger: the API can technically return all of these. https://github.com/home-assistant/core/blob/2d206e7e3172c68a191fb9487ca1370eb7958b15/homeassistant/helpers/typing.py#L14